.github/actions/spelling/patterns.txt

```diff
@@ -1,21 +1,22 @@
-[Ll][Oo][Mm][Bb][Ii][Qq]
-(\((.*?)\))
-(https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|www\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9]+\.[^\s]{2,}|www\.[a-zA-Z0-9]+\.[^\s]{2,})
+# Text entries with special casing.
+[Cc](loud|LOUD)[Ss](mith|MITH)
 [Hh]ome[Pp]age
-[Dd]otnet
-[Ll][Ii][Nn][Qq]
-[Ll]ocalhost
-[Mm][Vv][Cc]
-[Uu]ncomment
-[Jj]ira
-[Ii][Ii][Ss]
-[Cc]mdlet
-[Aa]tata
-[Cc][Ll][Oo][Uu][Dd][Ss][Mm][Ii][Tt][Hh]
-[Ss][Dd][Kk]
-[Rr][Ee][Aa][Dd][Mm][Ee]
-[Oo][Ss][Oo][Cc][Ee]
-[Ss][Ff][Cc]
+[Xx][Pp]ath
 [Xx][Uu]nit
-[Xx][Pp][Aa][Tt][Hh]
-[Rr]eadme
+
+(\((.*?)\))
+
+# URI/URL formats.
+(https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|www\.[a-zA-Z0-9][a-zA-Z0-9-]+[a-zA-Z0-9]\.[^\s]{2,}|https?:\/\/(?:www\.|(?!www))[a-zA-Z0-9]+\.[^\s]{2,}|www\.[a-zA-Z0-9]+\.[^\s]{2,})
```

Comment:
I've committed your suggestion above, thanks!
1. I'm not sure, @Luko6 or @Piedone can answer maybe if there are edge cases we need to consider here.
2. I agree about simplification, even making a mistake like that would be detected with simple testing during development (i.e., does the link I pasted work?). Can you please rework these, based on your judgement, in a way that they are easy to read (and maintain, if necessary), but remain functional. Please also add examples in a comment what use case(s) they cover.

Reply:
None of the above would allow localhost urls either.

Let's remember that this is only about spell-checking and ignoring string that look like URLs. So maybe simply:
```suggestion
https?:\/\/[^\s]+
www.[^\s]+
```
That will match string starting with `http[s]://` or `www.` until the first space character.

Reply:
I'm not sure what I can provide to the conversation about URL patterns :). Please look up a commonly used pattern (note that this should be perl regex here) and use that.

Also, note, that the patterns file is not about matching arbitrary strings, but lines: If a line matches the pattern, then the whole line will be excluded from spell-checking.

Reply:
I actually didn't know that patterns.txt ignores the whole line where it matches, that makes the recently removed "text in parenthesis" pattern and the other, single-word entries even more problematic. I'll remove the rest of the single-word entries too, then - e.g., `[Hh]ome[Pp]age` in patterns can be substituted with `homepage` and `homePage` in allow.txt, but I'll test spelling without these entries to see which casing combinations we need to actually cover.